src/inference/enumerate.ad.js

```diff
@@ -47,6 +62,17 @@ module.exports = function(env) {
     return this.wpplFn(_.clone(this.store), env.exit, this.a);
   };
 
+  // Error function for error handling
+  // this.throwOnError is true: directly throw error
+  // this.throwOnError is false: return error (string) as infer result
+  Enumerate.prototype.error = function(errType) {
+    if (this.throwOnError) {
+      throw new Error(errType);
+    } else {
+      return this.k(this.store, errType + '..quit enumerate');
```

Comment:
I think that these `error` methods should be general purpose. The `".. quit enumerate"` string seems like something specific to "default infer", so I wonder if it shouldn't be somewhere else? Perhaps as part of `DefaultInfer` in `header.wppl`?